controllers/stockControllers.go

```diff
@@ -0,0 +1,337 @@
+	}
+
+	// Create the user in the database
+	user := models.User{Username: requestBody.Username, Email: requestBody.Email, Balance: requestBody.Balance, Password: requestBody.Password}
+
+	if err := initializers.DB.Create(&user).Error; err != nil {
+		c.JSON(500, gin.H{"error": "Failed to create user"})
+		return
+	}
+
+	c.JSON(201, gin.H{"message": "User created successfully"})
+}
+
+func Login(c *gin.Context) {
+	var user struct {
+		Username string `form:"username"`
+		Password string `form:"password"`
+	}
+
+	if err := c.ShouldBind(&user); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	var foundUser models.User
+	initializers.DB.Where("username = ?", user.Username).First(&foundUser)
+
+	claims := models.JWTClaims{
+		StandardClaims: jwt.StandardClaims{
+			ExpiresAt: time.Now().Add(time.Hour * 1).Unix(),
+		},
+		UserID: foundUser.ID,
+	}
+
+	token, err := jwt.NewWithClaims(jwt.SigningMethodHS256, claims).SignedString([]byte("secret"))
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Could not generate token"})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"token": token})
+}
+
+func UserData(c *gin.Context) {
+	username := c.Param("username")
+	// Check if the user data is cached in Redis
+	cachedUserData, err := initializers.RedisClient.Get(context.Background(), username).Result()
+	if err == nil {
+		// Data found in cache, return it
+
+		var cachedUser models.User
+		// Unmarshal the cached user data from JSON
+		if err := json.Unmarshal([]byte(cachedUserData), &cachedUser); err != nil {
+			// Handle unmarshaling error
+			c.JSON(500, gin.H{
+				"error": "Failed to unmarshal cached user data",
+			})
+			return
+		}
+
+		c.JSON(200, gin.H{
+			"Data": cachedUserData,
+		})
+		return
+	}
+	// Data not found in cache, fetch from the database
+	var user models.User
+	result := initializers.DB.First(&user, "username = ?", username)
+	if result.Error != nil {
+		c.JSON(404, gin.H{
+			"error": "User not found",
+		})
+		return
+	}
+	// Cache the user data in Redis for future requests
+	err = initializers.RedisClient.Set(context.Background(), username, user, 5*time.Minute).Err()
+	if err != nil {
+		// Handle Redis cache error
+		c.JSON(500, gin.H{
+			"error": "Failed to cache user data",
+		})
+		return
+	}
+
+	c.JSON(200, gin.H{
+		"user": user,
+	})
+}
+
+func IngestStockData(c *gin.Context) {
+	var body struct {
+		Ticker     string
+		OpenPrice  float64
+		ClosePrice float64
+		High       float64
+		Low        float64
+		Volume     int
+	}
+
+	if err := c.Bind(&body); err != nil {
+		// Handle the error (e.g., return an error response)
+		return
+	}
+	data := models.StockData{Ticker: body.Ticker, OpenPrice: body.OpenPrice, ClosePrice: body.ClosePrice, High: body.High, Low: body.Low, Volume: body.Volume}
+	result := initializers.DB.Create(&data)
+
+	if result.Error != nil {
+		c.Status(400)
+		return
+	}
+	c.JSON(200, gin.H{
+		"mesage": "data inserted successfully",
```

Comment:
Add all the messages in a separate constants file and import from there.
```suggestion
		"mesage": "DATA_SAVED_SUCCESSFULLY",
```
DATA_SAVED_SUCCESSFULLY=data inserted successfully

Reply:
I resolved this issue